lua/starfall/libs_sh/entities.lua

```diff
@@ -300,50 +300,54 @@ function ents_methods:getLinkedComponents()
 -- @param Entity? parent Entity parent (nil to unparent)
--- @param number|string? attachment Optional attachment name or ID
+-- @param number|string? attachment Optional attachment name or ID.
 -- @param number|string? bone Optional bone name or ID. Can't be used at the same time as attachment
 function ents_methods:setParent(parent, attachment, bone)
 	local child = getent(self)
-	if CLIENT and debug.getmetatable(child)~=SF.Cl_Hologram_Meta then SF.Throw("Can only setParent clientside holograms in the clientside!", 2) end
 	checkpermission(instance, child, "entities.setParent")
-	if attachment~=nil and bone~=nil then SF.Throw("Can't have both attachment and bone args set!", 2) end
+	if CLIENT and debug.getmetatable(child) ~= SF.Cl_Hologram_Meta then SF.Throw("Only clientside holograms can be parented in the CLIENT realm!", 2) end
+	if attachment ~= nil and bone ~= nil then SF.Throw("Arguments `attachment` and `bone` are mutually exclusive!", 2) end
+	
 	if parent ~= nil then
 		parent = getent(parent)
-		if parent:IsPlayer() and not child.IsSFHologram then SF.Throw("Can only setParent holograms onto players!", 2) end
-		local param, type
-		if bone~=nil then
+		if parent:IsPlayer() and not child.IsSFHologram then SF.Throw("Only holograms can be parented to players!", 2) end
+		
+		local type, param
+		if bone ~= nil then
 			if isstring(bone) then
 				bone = parent:LookupBone(bone) or -1
 			elseif not isnumber(bone) then
 				SF.ThrowTypeError("string or number", SF.GetType(bone), 2)
 			end
-			if bone < 0 or bone > 255 then SF.Throw("Invalid bone provided", 2) end
+			if bone < 0 or bone > 255 then SF.Throw("Invalid bone provided!", 2) end
+			
 			type = "bone"
 			param = bone
-		elseif attachment~=nil then
-			if CLIENT then SF.Throw("Can't use attachment argument in clientside!", 2) end
-			if isstring(attachment) then
-				attachment = parent:LookupAttachment(attachment)
-			elseif not isnumber(attachment) then
+		elseif attachment ~= nil then
+			if CLIENT then SF.Throw("Parenting to an attachment is not supported in clientside!", 2) end
+			if isnumber(attachment) then
+				local attachments = parent:GetAttachments()
+				if attachments and attachments[attachment] then
+					attachment = attachments[attachment].name
+				else
+					SF.Throw("Invalid attachment ID provided!", 2)
+				end
+			elseif not isstring(attachment) then
 				SF.ThrowTypeError("string or number", SF.GetType(attachment), 2)
 			end
-			if attachment < 0 or attachment > 255 then SF.Throw("Invalid attachment provided", 2) end
+			if parent:LookupAttachment(attachment) < 1 then SF.Throw("Invalid attachment provided!", 2) end
+			
 			type = "attachment"
 			param = attachment
 		else
 			type = "entity"
 		end
-
-		SF.Parent(parent, child, type, param)
+		
+		SF.Parent(child, parent, type, param)
 	else
-		local sf_parent = child.sf_parent
-		if sf_parent then
-			sf_parent:setParent()
-		else
-			child:SetParent()
-		end
```

Comment:
This reintroduces the bug of not being able to unparent things not parented using starfall.

Reply:
Fixed. It's now in `SF.Parent`'s `__call`. I think that's a better place for it than `Entity.setParent` method.